### PR TITLE
Adds author

### PR DESCRIPTION
**Pull request by isamitre**
adds user authentication to UserPage
adds authorID to Recipe entity
lists user recipes by filtering the Recipe entity by the authorID
it does make you re-login every time the page reloads though

### CODE REVIEW

File: src/main/java/com/google/sps/data/Recipe.java [hunk 1]
```diff
@@ -23,16 +23,18 @@ public Recipe(long id, String name, String imageBlobKey, String description) {
...
     tags = new ArrayList<String>();
     published = false;
   }
 
-  public Recipe(long id, String name, String imageBlobKey, String description, ArrayList<String> tags, 
+  public Recipe(long id, String name, String authorID, String imageBlobKey, String description, ArrayList<String> tags, 
       ArrayList<String> ingredients, ArrayList<String> steps, boolean published, long popularity) {
     this.id = id;
     this.name = name;
+    this.authorID = authorID;
```
Comment (ez-may) [inline src/main/java/com/google/sps/data/Recipe.java:37]: Nit: are we not using the convention on `authorId` instead of `authorID`?

**Review by ez-may — APPROVED** [review]
LGTM

File: src/main/java/com/google/sps/data/Recipe.java (hunk 1 above)
Comment (isamitre) [reply src/main/java/com/google/sps/data/Recipe.java:37]: the methods/functions we keep in camelcase (i.e. getAuthorId, getId) but the variable names we said could stay as ID

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java [hunk 2]
```diff
@@ -46,14 +48,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
+    User user = new User(idToken);
+
     Entity recipeEntity = new Entity("Recipe");
     recipeEntity.setProperty("name", name);
     recipeEntity.setProperty("tags", tags);
     recipeEntity.setProperty("description", description);
     recipeEntity.setProperty("ingredients", ingredients);
     recipeEntity.setProperty("steps", steps);
+    recipeEntity.setProperty("authorID", user.getId());
```
Comment (oturner24) [inline src/main/java/com/google/sps/servlets/NewRecipeServlet.java:59]: Can you do `recipeEntity.setProperty("authorID", idToken)`? (and not create a User) Or is this id different?

**Review by oturner24 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java (hunk 2 above)
Comment (isamitre) [reply src/main/java/com/google/sps/servlets/NewRecipeServlet.java:59]: the id is different; I have to get it from the idToken
to be honest, I'm just doing what Ali told me; he said to get the id by making a new User

**Pull request merged**